django_cloud_deploy/cli/prompt_manager.py

```diff
@@ -0,0 +1,673 @@
+            self.console.error(e)
+            return False
+
+        return False
+
+
+class StringTemplatePromptManager(TemplatePromptManager):
+    """Template for a simple string Prompt Manager."""
+
+    PARAMETER = ''
+    PARAMTER_PRETTY = ''
+    DEFAULT_VALUE = ''
+    BASE_MESSAGE = '{} Enter a value for {} or leave blank to use'
+    DEFAUlT_MESSAGE = '[{}]: '
+
+    def _generate_name_prompt(self):
+        base_message = self.BASE_MESSAGE.format(self.step, self.PARAMTER_PRETTY)
+        default_message = self.DEFAUlT_MESSAGE.format(self.DEFAULT_VALUE)
+        msg = '\n'.join([base_message, default_message])
+        return prompt_new.AskPrompt(
+            msg, self.console, self.validate, default=self.DEFAULT_VALUE)
+
+    def prompt(self):
+        if self._is_valid_passed_arg():
+            return
+
+        self.args[self.PARAMETER] = self._generate_name_prompt().prompt()
+
+
+class GoogleProjectNameManager(TemplatePromptManager):
+
+    PARAMETER = 'project_name'
+
+    def __init__(self,
+                 console: io.IO,
+                 step: str,
+                 args,
+                 project_client: Optional[project.ProjectClient] = None):
+        super().__init__(console, step, args)
+        creds = self.get_credentials()
+        self.project_client = (project_client or
+                               project.ProjectClient.from_credentials(creds))
+
+    def validate(self, s: str) -> bool:
+        """Validates that a string is a valid project name.
+
+        Args:
+            s: The string to validate.
+
+        Raises:
+            ValueError: if the input string is not valid.
+        """
+        if not (4 <= len(s) <= 30):
+            raise ValueError(
+                ('Invalid Google Cloud Platform project name "{}": '
+                 'must be between 4 and 30 characters').format(s))
+
+        if self._is_new_project():
+            return True
+
+        project_id = self.args.get('project_id', None)
+        if project_id is None:
+            raise ValueError('Project Id must be set')
+
+        project_name = self.project_client.get_project(project_id)['name']
+        if project_name != s:
+            raise ValueError('Wrong project name given for project id.')
+
+        return True
+
+    def _generate_name_prompt(self):
+        default_answer = 'Django Project'
+        msg_base = ('{} Enter a Google Cloud Platform project name, or leave '
+                    'blank to use').format(self.step)
+        msg_default = '[{}]: '.format(default_answer)
+        msg = '\n'.join([msg_base, msg_default])
+        return prompt_new.AskPrompt(
+            msg, self.console, self.validate, default=default_answer)
+
+    def _is_new_project(self):
+        must_exist = workflow.ProjectCreationMode.MUST_EXIST
+        return self.args.get('project_creation_mode', None) != must_exist
+
+    def _handle_existing_project(self):
+        assert 'project_id' in self.args, 'project_id must be set'
```

Comment:
This can't be an assert, right? Because the user could really pass

 --project_creation_mode=MUST_EXIST but not specify a project_id.

Reply:
Project creation mode is hard coded to create unless they pass in a project-id.

Reply:
In other words, its not a flag available to the user.

Reply:
I think this would be more of a safeguard for ourselves as there is not reason it should reach this branch if project_id is not already set by this point. (as well as project_id being prompted before project_name)